velox/functions/prestosql/aggregates/tests/MinMaxByAggregationTest.cpp

```diff
@@ -1340,25 +1305,108 @@ TEST_F(MinMaxByComplexTypes, rowCompare) {
       }),
   });
 
-  VELOX_ASSERT_THROW(
-      testAggregations(
-          {data}, {}, {"min_by(c0, c1)", "max_by(c0, c1)"}, {expected}),
-      "ROW comparison not supported for values that contain nulls");
+  testAggregations(
+      {data}, {}, {"min_by(c0, c1)", "max_by(c0, c1)"}, {expected});
+}
 
-  data = makeRowVector({
-      makeArrayVector<int64_t>({
-          {1, 2, 3},
-          {4, 5},
-          {6, 7, 8},
+TEST_F(MinMaxByComplexTypes, arrayCheckNulls) {
+  auto batch = makeRowVector({
+      makeFlatVector<int32_t>({
+          1,
+          2,
+          3,
       }),
-      makeRowVector({makeNullableFlatVector<int32_t>({
+      makeArrayVectorFromJson<int32_t>({
+          "[1, 2]",
+          "[6, 7]",
+          "[2, 3]",
+      }),
+      makeFlatVector<int32_t>({
           1,
           2,
           3,
-      })}),
+      }),
   });
-  testAggregations(
-      {data}, {}, {"min_by(c0, c1)", "max_by(c0, c1)"}, {expected});
+
+  auto batchWithNull = makeRowVector({
+      makeFlatVector<int32_t>({
+          1,
+          2,
+          3,
+      }),
+      makeArrayVectorFromJson<int32_t>({
+          "[1, 2]",
+          "[6, 7]",
+          "[3, null]",
+      }),
+      makeFlatVector<int32_t>({
+          1,
+          2,
+          3,
+      }),
+  });
+
+  for (const auto& expr : {"min_by(c0, c1)", "max_by(c0, c1)"}) {
+    testFailingAggregations(
+        {batch, batchWithNull},
+        {},
+        {expr},
+        "ARRAY comparison not supported for values that contain nulls");
+    testFailingAggregations(
+        {batch, batchWithNull},
+        {"c2"},
+        {expr},
+        "ARRAY comparison not supported for values that contain nulls");
+  }
+}
+
+TEST_F(MinMaxByComplexTypes, rowCheckNull) {
+  auto batch = makeRowVector({
+      makeFlatVector<int8_t>({1, 2, 3}),
+      makeRowVector({
+          makeFlatVector<StringView>({
```

Comment:
`makeFlatVector<std::string>({"a", "b", "c"})` for readability

ditto other places

Reply:
Fixed